share/scripts/inverse/run_gromacs.sh

```diff
@@ -115,12 +117,32 @@ if [[ ${CSG_MDRUN_OPTS} ]]; then
   mdrun_opts+=" ${CSG_MDRUN_OPTS}"
 fi
 
+#support of REMD during gromacs simlation
+if [[ $(csg_get_property cg.inverse.gromacs.REMD) = "yes" ]]; then
+PREFIX="$(csg_get_property cg.inverse.gromacs.REMD.PREFIX)"
+read -a T <<<"$(csg_get_property cg.inverse.gromacs.REMD.T_for_replicas)"
+j=0
+for i in "${T[@]}"
+do
+echo $i
+#TEMP has to be written as a place holer for the temperature in the *.mdp file
+sed "s/TEMP/$i/" grompp.mdp > grompp_$j.mdp
+critical ${grompp[@]} -n "${index}" -f "grompp_$j.mdp" -p "$topol_in" -o "$PREFIX"_"$j.tpr" -c "${conf}" ${grompp_opts} 2>&1 | gromacs_log "${grompp[@]} -n "${index}" -f "${mdp}" -p "$topol_in" -o "$tpr" -c "${conf}" ${grompp_opts}"
```

Comment:
Would it be too much effort for the user to make `grompp_1.mdp` ... `grgrompp_n.mdp` by hand and list them in `filelist` explicitly? (And the same for `conf*.gro`?)

@DRosen285 , @abrukhno : Opinions?

Reply:
Hi David, thanks for your insights. The link does not describe the workflow completely, so I can only assume that:

(a) all the input files must be repeated N_replicas times, their names taking the form <file-name>_<replica-index>.<extension>, as it is for '-multi' option (which is anyway/always needed for REMD, apparently); 
NOTE that, according to another description: http://manual.gromacs.org/documentation/2016.3/user-guide/mdrun-features.html , one can use '-multidir' instead of '-multi', in which case the user has to prepare N_replicas (indexed) subdirectories that should contain input files for each replica separately, and no indexing in file names is needed, i.e. one can use grompp.mdp, config.mdp topol.top etc which may contain different inputs.

(b) correspondingly, the output will comprise N_replicas of output files named according to the chosen option, either '-multi' or '-multidir', in the latter case the output files for each replica are stored in the corresponding (indexed) subdirectories. 

(c) post-simulation analysis should either extract/combine the trajectories per ensemble (a single trajectory per T value) or per replicas time (a single trajectory per replica, where T is varying), or both, depending on the user's task/requirements.

Based on the above, I suggest that we do not split this workflow into REMD/multi branches (as suggested by different 'runmode's), but rather combine the two within the same branch. The tags might work like this: 'cg.inverse.gromacs.replicas'=number, >0 for '-multi'/'-multidir' options (default=0, meaning normal mdrun run), and 'cg.inverse.gromacs.replicas.ref-ts'=string (default ="", i.e. empty), which for REMD won't be empty and contain a set of T-values. 

Then the user will have full control over the natural Gromacs workflows and have freedom to decide on the bookkeeping wrt input and output files etc. We will have an extra complication, though, to arrange the concatenation of trajectory file according to another option: 'cg.inverse.gromacs.replicas.trjcat_opts'

What do you think? - Christoph, you are welcome to comment too!

Reply:
sorry, the internal parsing here ate my input for file names: file-name_index.extension (I tried to use angle brackets above, for no avail).